alpha/apps/kaltura/lib/batch2/kFlowHelper.php

```diff
@@ -3293,16 +3293,16 @@ public static function addPeriodicStorageExports($entryId, $partner, $periodicSt
 		}
 	}
 
-	protected static function checkNonPeriodicExportsFinished($partner, $assetsIds)
+	public static function checkNonPeriodicExportsFinished($partnerId, $assetsIds)
 	{
-		$storageProfiles = StorageProfilePeer::retrieveExternalByPartnerId($partner->getId());
+		$storageProfiles = StorageProfilePeer::retrieveExternalByPartnerId($partnerId);
 		$status = array(FileSync::FILE_SYNC_STATUS_PENDING, FileSync::FILE_SYNC_STATUS_ERROR);
 		$types = array(FileSyncObjectType::ASSET);
 		$subTypes = array(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
 		$allFinished = true;
 		foreach($storageProfiles as $profile)
 		{
-			$fileSyncs = FileSyncPeer::retrieveFileSyncsByFlavorAndDc($profile->getId(), $partner->getId(), $status, $assetsIds, $types, $subTypes);
+			$fileSyncs = FileSyncPeer::retrieveFileSyncsByFlavorAndDc($profile->getId(), $partnerId, $status, $assetsIds, $types, $subTypes);
```

Comment:
it means that if a partner has remote storage stuck in error , the content will never be push to periodic.
For that reason we should issue a warning message , in case we find file sync in error.